pkg/controller/migmigration/stage.go

```diff
@@ -340,9 +343,19 @@ func (t *Task) stagePodLabels() map[string]string {
 }
 
 // Build a stage pod based on existing pod.
-func buildStagePodFromPod(ref k8sclient.ObjectKey, labels map[string]string, pod *corev1.Pod) StagePod {
+func buildStagePodFromPod(ref k8sclient.ObjectKey, labels map[string]string, pod *corev1.Pod) *StagePod {
+	pvcVolumes := []corev1.Volume{}
+	for _, volume := range pod.Spec.Volumes {
+		if volume.PersistentVolumeClaim != nil {
+			pvcVolumes = append(pvcVolumes, volume)
```

Comment:
To avoid creating stage pods when we're not using restic for any volumes, it might be worth filtering based on the migplan's PV list here as well. The hard part is that this is logic we're already going through in the set annotations phase, and we need to avoid duplicating too much or this may end up being too slow, but we really need similar filtering here. We only want to include volumes that:
1) are PVC volumes
2) action=copy
3) copyMethod=filesystem

Reply:
Added. Please review.